### ISSUE TITLE
Upgrade to Tornado 6

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-uiserver#9]
https://www.tornadoweb.org/en/stable/releases/v6.0.0.html

Might be a good idea to freeze pip and use specific versions in the dependencies list too.

**Comment by kinow** [comment]
This might be blocked until JupyterHub updates too. But I recall seeing a maintainer from JupyterHub testing 6.0 and providing feedback about a bug they found in the websockets part. So it should happen there too soon.

**Comment by dwsutherland** [comment]
We are already running JupyterHub with `6.0.2` and we're getting no deprecation warnings from Tornado when running with the flag (as Tornado suggest in the release notes):
```
(uis-test) sutherlander@cortex-vbox:cylc-uiserver$ export PYTHONWARNINGS=d
(uis-test) sutherlander@cortex-vbox:cylc-uiserver$ jupyterhub
/home/sutherlander/.envs/uis-test/lib/python3.7/site-packages/jinja2/utils.py:485: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from collections import MutableMapping
/home/sutherlander/.envs/uis-test/lib/python3.7/site-packages/jinja2/runtime.py:318: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from collections import Mapping
[I 2019-04-15 23:54:14.764 JupyterHub app:1673] Using Authenticator: jupyterhub.auth.PAMAuthenticator-0.9.6
[I 2019-04-15 23:54:14.765 JupyterHub app:1673] Using Spawner: jupyterhub.spawner.LocalProcessSpawner-0.9.6
[I 2019-04-15 23:54:14.766 JupyterHub app:1016] Loading cookie_secret from /home/sutherlander/repos/kinow/cylc-uiserver/jupyterhub_cookie_secret
/home/sutherlander/.envs/uis-test/lib/python3.7/site-packages/sqlalchemy/engine/strategies.py:144: SADeprecationWarning: PoolListener is deprecated in favor of the PoolEvents listener interface.  The Pool.listeners parameter will be removed in a future release.
  pool = poolclass(creator, **pool_args)
/home/sutherlander/.envs/uis-test/lib/python3.7/site-packages/sqlalchemy/pool/base.py:220: SADeprecationWarning: The Pool.add_listener() method is deprecated and will be removed in a future release.  Please use the PoolEvents listener interface.
  self.add_listener(l)
/home/sutherlander/.envs/uis-test/lib/python3.7/site-packages/sqlalchemy/pool/base.py:281: SADeprecationWarning: PoolListener.connect is deprecated.  The PoolListener class will be removed in a future release.  Please transition to the @event interface, using @event.listens_for(Engine, 'connect').
  interfaces.PoolListener._adapt_listener(self, listener)
```
JupyterHub only has `>=5.0` in their [requirements](https://github.com/jupyterhub/jupyterhub/blob/master/requirements.txt).

What do they have left to do?

**Comment by kinow** [comment]
>What do they have left to do?

I think I created this issue shortly after Tornado 6 releases came out. There was at least one bug with Tornado and JupyterHub, if I remember well, it was something with WebSockets. I checked and I think they had not implicitly updated to `>=6.0`...

Right now I wonder if their codebase is still compatible with 5.x, or if it's only 6.x. If the latter, then we can prepare a pull request there.
